ext/standard/basic_functions.c

```diff
@@ -1310,8 +1310,8 @@ PHP_FUNCTION(time_sleep_until)
 	target_ns = (uint64_t) (target_secs * ns_per_sec);
 	current_ns = ((uint64_t) tm.tv_sec) * ns_per_sec + ((uint64_t) tm.tv_usec) * 1000;
 	if (target_ns < current_ns) {
-		php_error_docref(NULL, E_WARNING, "Sleep until to time is less than current time");
-		RETURN_FALSE;
+		zend_argument_value_error(1, "must be a timestamp greater than or equal to the current time");
+		RETURN_THROWS();
```

Comment:
This seems a bit risky given potential clock changes (or bugs -- macos has incredible timers!)

Reply:
Reverting the exception change sounds reasonable.

P.S. I think the biggest risk is to use this function when DST begins (in a time zone observing DST). And can't even a short `set_time_limit()` setting help. :D